### ISSUE TITLE
kops 1.7 performs uneccesary rolling update even when no changes.

### ISSUE THREAD

**Issue by prateek-1708** [kubernetes/kops#3450]
`kops` incorrectly thinks that the cluster needs to be updated every single time ever since we moved from `kops` `1.6` to `kops` `1.7` even though there is no change in the source code other than few `.md`s
0
In the plan stage it shows following 
```
  DNSName/api.k8s.banana.com
  	Lifecycle           	 <nil> -> Sync

  InternetGateway/k8s.banana.com
  	Name                	 dev -> k8s.banana.com

  LaunchConfiguration/master-a.masters.k8s.banana.com
  	RootVolumeOptimization	 <nil> -> true

  LaunchConfiguration/master-b.masters.k8s.banana.com
  	RootVolumeOptimization	 <nil> -> true

  LaunchConfiguration/master-c.masters.k8s.banana.com
  	RootVolumeOptimization	 <nil> -> true

  LaunchConfiguration/master-d.masters.k8s.banana.com
  	RootVolumeOptimization	 <nil> -> true

  LaunchConfiguration/master-e.masters.k8s.banana.com
  	RootVolumeOptimization	 <nil> -> true

  LaunchConfiguration/nodes.k8s.banana.com
  	RootVolumeOptimization	 <nil> -> true

  LoadBalancer/api.k8s.banana.com
  	Lifecycle           	 <nil> -> Sync

  LoadBalancerAttachment/api-master-a
  	Lifecycle           	 <nil> -> Sync

  LoadBalancerAttachment/api-master-b
  	Lifecycle           	 <nil> -> Sync

  LoadBalancerAttachment/api-master-c
  	Lifecycle           	 <nil> -> Sync

  LoadBalancerAttachment/api-master-d
  	Lifecycle           	 <nil> -> Sync

  LoadBalancerAttachment/api-master-e
  	Lifecycle           	 <nil> -> Sync
......
 ```

What confuses me is
* we aready have `rootVolumeOptimization: true` in `kops` cluster definition then why does it think that it needs to update ?
* even if it wants to ignore the flag we have set, why would `kops` do it every single time ? 

Please let us know if there is a fix for this or if it is feature ?

**Comment by gambol99** [comment]
Hi @prateek-1708 ... I'm not sure about the LaunchConfiguration but the LoadBalancerAttachment was fixed on master https://github.com/kubernetes/kops/pull/3226. .. I'll have a look a LaunchConfiguration

**Comment by prateek-1708** [comment]
That would be super awesome ... thanks for the update @gambol99

**Comment by Pharb** [comment]
@prateek-1708 the problem with the RootVolumeOptimization has been fixed on master, see #3313.

Until the next kops version is released, you can ignore the required changes from `RootVolumeOptimization	 <nil> -> true` by not starting the rolling update, or build your own kops version that includes the bugfix.

**Comment by chrislovecnm** [comment]
@Pharb alpha is out as well, stability is not guaranteed as of yet, but they do not have to compile from source

**Comment by chrislovecnm** [comment]
alpha kops version

**Comment by chrislovecnm** [comment]
@prateek-1708 can we close this since this is fixed in master?

**Comment by prateek-1708** [comment]
@chrislovecnm  any idea if this would be patched to 1.7 release ?? or do we have to wait for 1.8 release ? and would that be 1.7 compatible ??

**Comment by chrislovecnm** [comment]
@prateek-1708 1.8.0 alpha is out, and yes we are backward compatible with kubernetes.
